### ISSUE TITLE
`SIGSEGV` during initial sync of Polygon zkEVM

### ISSUE THREAD

**Issue by yorickdowne** [0xPolygon/cdk-erigon#944]
#### System information

Erigon `1.2.15.2`
```
[cdk-erigon-lib] timestamp 2024-03-12:16:34
cdk-erigon version 2.43.0-dev-70544f8e
```

Linux Debian 12

Docker 27.1.1

Erigon Command (with flags/config):

`cdk-erigon --config /home/erigon/.local/share/erigon-config/mainnet.yaml --zkevm.l1-rpc-url https://my-redacted-rpc.example.com --zkevm.rpc-ratelimit=250 --maxpeers 32 --downloader.disable.ipv6 --http --http.api eth,net,trace,web3,erigon,zkevm --http.addr 0.0.0.0 --http.port 8545 --http.vhosts=* --http.corsdomain=* --ws --metrics --metrics.addr 0.0.0.0`

Chain/Network: 

Polygon zkEVM

#### Issue

During initial sync of Polygon zkEVM, Erigon failed with a `SIGSEGV`, after finishing 3/16 Batches "the second time" ... it goes through execution and 9/16 etc, and then comes to 3/16 again, and fails.

```
zkevm-erigon-1  | [INFO] [08-09|01:11:19.014] [3/16 Batches] Finished Batches stage 
zkevm-erigon-1  | [INFO] [08-09|01:11:19.014] [3/16 Batches] DONE                      in=5m19.963479648s
zkevm-erigon-1  | panic: runtime error: invalid memory address or nil pointer dereference
zkevm-erigon-1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x8f9843]
zkevm-erigon-1  | 
zkevm-erigon-1  | goroutine 134 [running, locked to thread]:
zkevm-erigon-1  | github.com/torquem-ch/mdbx-go/mdbx.openCursor.func1(0x0?, 0x1320b7c0?, 0xc0078318e0)
zkevm-erigon-1  |       github.com/torquem-ch/mdbx-go@v0.27.10/mdbx/cursor.go:69 +0x23
zkevm-erigon-1  | github.com/torquem-ch/mdbx-go/mdbx.openCursor(0x0, 0x13874d0?)
zkevm-erigon-1  |       github.com/torquem-ch/mdbx-go@v0.27.10/mdbx/cursor.go:69 +0x65
zkevm-erigon-1  | github.com/torquem-ch/mdbx-go/mdbx.(*Txn).OpenCursor(...)
zkevm-erigon-1  |       github.com/torquem-ch/mdbx-go@v0.27.10/mdbx/txn.go:597
zkevm-erigon-1  | github.com/gateway-fm/cdk-erigon-lib/kv/mdbx.(*MdbxTx).stdCursor(0xc00b8d9860, {0x268cffd, 0x9})
zkevm-erigon-1  |       github.com/gateway-fm/cdk-erigon-lib@v0.0.0-20240805074757-5e3196248fde/kv/mdbx/kv_mdbx.go:1075 +0x13f
zkevm-erigon-1  | github.com/gateway-fm/cdk-erigon-lib/kv/mdbx.(*MdbxTx).RwCursor(0xc00b8d9860, {0x268cffd, 0x9})
zkevm-erigon-1  |       github.com/gateway-fm/cdk-erigon-lib@v0.0.0-20240805074757-5e3196248fde/kv/mdbx/kv_mdbx.go:1062 +0xdb
zkevm-erigon-1  | github.com/gateway-fm/cdk-erigon-lib/kv/mdbx.(*MdbxTx).Cursor(...)
zkevm-erigon-1  |       github.com/gateway-fm/cdk-erigon-lib@v0.0.0-20240805074757-5e3196248fde/kv/mdbx/kv_mdbx.go:1066
zkevm-erigon-1  | github.com/gateway-fm/cdk-erigon-lib/kv/mdbx.(*MdbxTx).statelessCursor(0xc00b8d9860, {0x268cffd, 0x9})
zkevm-erigon-1  |       github.com/gateway-fm/cdk-erigon-lib@v0.0.0-20240805074757-5e3196248fde/kv/mdbx/kv_mdbx.go:919 +0xa9
zkevm-erigon-1  | github.com/gateway-fm/cdk-erigon-lib/kv/mdbx.(*MdbxTx).GetOne(0x0?, {0x268cffd?, 0xb?}, {0xc007946ea0, 0xb, 0x10})
zkevm-erigon-1  |       github.com/gateway-fm/cdk-erigon-lib@v0.0.0-20240805074757-5e3196248fde/kv/mdbx/kv_mdbx.go:945 +0x2d
zkevm-erigon-1  | github.com/ledgerwatch/erigon/eth/stagedsync/stages.GetStageProgress({0x2f1af60, 0xc00b8d9860}, {0x2690ade?, 0xc01320ba00?})
zkevm-erigon-1  |       github.com/ledgerwatch/erigon/eth/stagedsync/stages/stages.go:81 +0x55
zkevm-erigon-1  | github.com/ledgerwatch/erigon/eth/stagedsync.(*Sync).StageState(0xc0035b0580, {0x2690ade, 0xb}, {0x2f34340, 0xc00b8d9860}, {0x2f1ed08, 0xc0000e5b20})
zkevm-erigon-1  |       github.com/ledgerwatch/erigon/eth/stagedsync/sync.go:175 +0x99
zkevm-erigon-1  | github.com/ledgerwatch/erigon/eth/stagedsync.(*Sync).runStage(0xc0035b0580, 0xc0005e0370, {0x2f29598?, 0xc0000e5b20}, {0x2f415c0?, 0xc00b8d9860}, 0x42?, 0x96?, 0xf8?
)
zkevm-erigon-1  |       github.com/ledgerwatch/erigon/eth/stagedsync/sync.go:355 +0x11a
zkevm-erigon-1  | github.com/ledgerwatch/erigon/eth/stagedsync.(*Sync).Run(0xc0035b0580, {0x2f29598, 0xc0000e5b20}, {0x2f415c0, 0xc00b8d9860}, 0xa0?, 0x29?)
zkevm-erigon-1  |       github.com/ledgerwatch/erigon/eth/stagedsync/sync.go:262 +0x285
zkevm-erigon-1  | github.com/ledgerwatch/erigon/turbo/stages.StageLoopStep({0x2f16b40?, 0xc0005e0dc0}, 0x0?, {0x2f29598, 0xc0000e5b20}, 0xc0035b0580, 0xc0031c4fe0, 0x0, 0xc000bc6500)
zkevm-erigon-1  |       github.com/ledgerwatch/erigon/turbo/stages/stageloop.go:177 +0x46d
zkevm-erigon-1  | github.com/ledgerwatch/erigon/turbo/stages.StageLoop({0x2f16b40, 0xc0005e0dc0}, 0xc0006d4900?, {0x2f29598?, 0xc0000e5b20}, 0xc00122bfd0?, 0xc0017c2900, 0xc0000fc3c0?, 0x2bf6a18?, 0xc000154180, ...)
zkevm-erigon-1  |       github.com/ledgerwatch/erigon/turbo/stages/stageloop.go:97 +0x165
zkevm-erigon-1  | created by github.com/ledgerwatch/erigon/eth.(*Ethereum).Start
zkevm-erigon-1  |       github.com/ledgerwatch/erigon/eth/backend.go:1414 +0x225
zkevm-erigon-1  | [cdk-erigon-lib] timestamp 2024-03-12:16:34
zkevm-erigon-1  | [INFO] [08-09|01:11:20.507] logging to file system                   log dir=/home/erigon/.local/share/erigon/logs file prefix=erigon log level=info json=false
zkevm-erigon-1  | [INFO] [08-09|01:11:20.507] Starting metrics server                  addr=http://0.0.0.0:6060/debug/metrics/prometheus
```

#### Steps to reproduce the behaviour

Initial-sync Polygon zkEVM with 1.2.15.2. The SIGSEGV right after 3/16 Batches is happening consistently, even after restart.

**Comment by yorickdowne** [comment]
Testing the sync on `1.2.15.1`, with the DB created by `1.2.15.2`. mdbx complains about a file size mismatch: `mdbx_setup_dxb:15946 filesize mismatch (expect 100931731456b/24641536p, have 130996502528b/31981568p)`

**Comment by yorickdowne** [comment]
Solved by finishing the sync with `1.2.15.1`. Upgrade to `1.2.15.3` after was successful.

I do not know whether this issue also exists in `1.2.15.3`. As I didn't test this with the latest release, I am closing the issue. It can be re-opened if someone else encounters the same problem with `1.2.15.3` or later.
